packages/flame/lib/src/geometry/shape_intersections.dart

```diff
@@ -80,9 +80,10 @@ class CirclePolygonIntersections
       intersectionPoints.addAll(circle.lineSegmentIntersections(line));
     }
     if (intersectionPoints.isEmpty && (circle.isSolid || polygon.isSolid)) {
-      final outerShape = circle.containsPoint(polygon.vertices.first)
-          ? circle
-          : (polygon.containsPoint(circle.position) ? polygon : null);
+      final outerShape =
+          circle.containsPoint(polygon.vertices.first - polygon.topLeft)
```

Comment:
Yeah this is definitely not correct, how sloppy of me to not write a test for this case.

Reply:
Yup I can put together a PR for the circle-polygon intersection (will need to get up to speed on how the others work first though). Since this PR only maintains previous behaviour I agree merge should be good!